src/main/java/org/broadinstitute/hellbender/utils/text/ListFileUtils.java

```diff
@@ -0,0 +1,169 @@
+            throw new NullPointerException("values cannot be null");
+        Set<String> unpackedValues = new LinkedHashSet<String>();
+        // Let's first go through the list and see if we were given any files.
+        // We'll add every entry in the file to our set, and treat the entries as
+        // if they had been specified on the command line.
+        for (String value : values) {
+            File file = new File(value);
+            if (value.toLowerCase().endsWith(".list") && file.exists()) {
+                unpackedValues.addAll(new XReadLines(IOUtil.openFileForBufferedReading(file), true, LIST_FILE_COMMENT_START).readLines());
+            } else {
+                unpackedValues.add(value);
+            }
+        }
+        return unpackedValues;
+    }
+
+    /**
+     * Returns a new set of values including only values listed by filters
+     * <p/>
+     * Each element E of values can either be a literal string or a file.  For each E,
+     * we try to read a file named E from disk, and if possible all lines from that file are expanded
+     * into unique names.
+     * <p/>
+     * Filters may also be a file of filters.
+     *
+     * @param values     Values or files with values
+     * @param filters    Filters or files with filters
+     * @param exactMatch If true match filters exactly, otherwise use as both exact and regular expressions
+     * @return entries from values or the files listed in values, filtered by filters
+     */
+    public static Set<String> includeMatching(Collection<String> values, Collection<String> filters, boolean exactMatch) {
+        return includeMatching(values, IDENTITY_STRING_CONVERTER, filters, exactMatch);
+    }
+
+    /**
+     * Converts a type T to a String representation.
+     *
+     * @param <T> Type to convert to a String.
+     */
+    public static interface StringConverter<T> {
+        String convert(T value);
+    }
+
+    /**
+     * Returns a new set of values including only values matching filters
+     * <p/>
+     * Filters may also be a file of filters.
+     * <p/>
+     * The converter should convert T to a unique String for each value in the set.
+     *
+     * @param values     Values or files with values
+     * @param converter  Converts values to strings
+     * @param filters    Filters or files with filters
+     * @param exactMatch If true match filters exactly, otherwise use as both exact and regular expressions
+     * @return entries from values including only values matching filters
+     */
+    public static <T> Set<T> includeMatching(Collection<T> values, StringConverter<T> converter, Collection<String> filters, boolean exactMatch) {
+        if (values == null)
+            throw new NullPointerException("values cannot be null");
+        if (converter == null)
+            throw new NullPointerException("converter cannot be null");
+        if (filters == null)
+            throw new NullPointerException("filters cannot be null");
```

Comment:
A couple of general style comments that apply to all of this code:
- You can use Utils.nonNull to validate argument values.
- We generally use braces for all blocks, even one liners.
- We mark all variables and arguments final unless the can't be.

Reply:
These were verbatim from GATK3, but will try to look through this more closely.  i think when we get into VariantEval's code that's going to be a lot of this kind of stuff.